src/components/app/details/cIDetails/CIDetails.tsx

```diff
@@ -668,123 +671,106 @@ export const NoVulnerabilityViewWithTool = ({ scanToolId }: { scanToolId: number
-        if (artifactId) {
-            callGetSecurityIssues()
-        }
-    }, [artifactId])
+    const [scanResultLoading, scanResultResponse, scanResultError, reloadScanResult] = useAsync(
+        () => getSecurityScan({ artifactId, ...((isJobCI || isJobView) && { appId: computedAppId }) }),
+        [artifactId, computedAppId],
+        isSecurityScanV2Enabled,
+    )
+
+    const [executionDetailsLoading, executionDetailsResponse, executionDetailsError, reloadExecutionDetails] = useAsync(
+        () => getLastExecutionByAppArtifactId(artifactId, isJobCI || isJobView ? computedAppId : null),
+        [artifactId, computedAppId],
+        !isSecurityScanV2Enabled,
+    )
 
     const redirectToCreate = () => {
         if (!ciPipelineId) {
             return
         }
         push(
-            `${URLS.APP}/${appId ?? appIdFromParent}/${URLS.APP_CONFIG}/${URLS.APP_WORKFLOW_CONFIG}/${ciPipelineId}/${
+            `${URLS.APP}/${computedAppId}/${URLS.APP_CONFIG}/${URLS.APP_WORKFLOW_CONFIG}/${ciPipelineId}/${
                 URLS.APP_CI_CONFIG
             }/${ciPipelineId}/build`,
         )
     }
 
-    if (['failed', 'cancelled'].includes(status.toLowerCase()) || !artifactId ) {
+    if (['starting', 'running'].includes(status.toLowerCase())) {
+        return <CIRunningView isSecurityTab />
+    }
+
+    if (!artifactId || ['failed', 'cancelled'].includes(status.toLowerCase())) {
         return (
             <GenericEmptyState
                 title={EMPTY_STATE_STATUS.ARTIFACTS_EMPTY_STATE_TEXTS.NoArtifactsGenerated}
                 subTitle={EMPTY_STATE_STATUS.ARTIFACTS_EMPTY_STATE_TEXTS.NoArtifactsError}
             />
         )
     }
-    if (['starting', 'running'].includes(status.toLowerCase())) {
-        return <CIRunningView isSecurityTab />
-    }
-    if (securityData.isLoading) {
+
+    if (scanResultLoading || executionDetailsLoading) {
         return <Progressing pageLoader />
     }
-    if (securityData.isError) {
-        return <Reload />
+    if (scanResultError || executionDetailsError) {
+        return <Reload reload={isSecurityScanV2Enabled ? reloadScanResult : reloadExecutionDetails} />
     }
-    if (artifactId && !securityData.scanned) {
-        if (!securityData.scanEnabled) {
+    if (
+        (executionDetailsResponse && !executionDetailsResponse.result.scanned) ||
+        (scanResultResponse && !scanResultResponse.result.scanned)
+    ) {
+        if (!executionDetailsResponse?.result.scanEnabled) {
             return <ScanDisabledView redirectToCreate={redirectToCreate} />
         }
         return <ImageNotScannedView />
     }
-    if (artifactId && securityData.scanned && !securityData.vulnerabilities.length) {
-        return <NoVulnerabilityViewWithTool scanToolId={securityData.ScanToolId} />
+
+    const imageScanSeverities = scanResultResponse?.result.imageScan.vulnerability.summary.severities // For scan-result Api
+    const severityCount: SeverityCount = isSecurityScanV2Enabled
```

Comment:
This logic is duplicated in SecurityVulnerabilityCard can move this into a method to recieve severityCount and totalSeverities

Reply:
I believe we can abstract this logic even more